src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx

```diff
@@ -40,14 +46,17 @@ export function DataSourceMenu(props: DataSourceMenuProps): ReactElement | null
     dataSourceCallBackFunc,
     showDataSourceSelectable,
     disableDataSourceSelectable,
+    showDataSourceAggregatedView,
     fullWidth,
     hideLocalCluster,
     selectedOption,
     showDataSourceView,
-    filterFn,
+    dataSourceFilter,
+    activeDataSourceIds,
+    displayAllCompatibleDataSources,
   } = props;
 
-  if (!showDataSourceSelectable && !showDataSourceView) {
+  if (!showDataSourceSelectable && !showDataSourceView && !showDataSourceAggregatedView) {
     return null;
   }
```

Comment:
These 3 boolean values are used to control the appearance of this component. If it was just one variable, it would have been acceptable but anything more than one variable controlling how _exclusively_ something is rendered is wrong. This has to be changed to a single ENUM variable.

Reply:
Thanks for the comment! Will address this in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6239 if the proposed change looks good @AMoo-Miki